pilot/pkg/networking/core/v1alpha3/gateway_test.go

```diff
@@ -1566,6 +1566,25 @@ func TestBuildGatewayListeners(t *testing.T) {
 			},
 			[]string{"0.0.0.0_80", "0.0.0.0_8080"},
 		},
+		{
+			"gateway with bind",
+			&pilot_model.Proxy{},
+			&networking.Gateway{
+				Servers: []*networking.Server{
+					{
+						Port: &networking.Port{Name: "http", Number: 80, Protocol: "HTTP"},
+					},
+					{
+						Port: &networking.Port{Name: "http", Number: 8080, Protocol: "HTTP"},
+					},
+					{
+						Port: &networking.Port{Name: "http", Number: 8080, Protocol: "HTTP"},
+						Bind: "127.0.0.1",
+					},
+				},
+			},
+			[]string{"0.0.0.0_80", "127.0.0.1_8080"},
```

Comment:
This doesn't look right. Shouldn't we have 0.0.0.0_8080?

Reply:
I think this should be prevented by validation

Reply:
It could happen from 2 different gateway objects bind to the same port but different bind. We should make sure we handle this well

Reply:
So here is the expectation that if a server does not specify bind (wild card) and server specifies bind on same port - there should two listeners with serving hosts from the respective servers?